### DISCUSSION TITLE
Dependency on Node.js?

### DISCUSSION THREAD

**Discussion by peteraldous** [scala/scala-parser-combinators#461] [Q&A] [answered]
I'm working on a pull request. I finished a first pass at the feature I'd like to add and ran `sbt test` to make sure I didn't break anything before writing some additional tests of my own.

I didn't mind installing clang (I've been meaning to start a project that uses LLVM this summer), but I'm more than a little surprised that this test suite depends on Node.js. I'm mostly unfamiliar with Node, although I'm aware of the left pad fiasco and don't trust the framework; because of my unfamiliarity, I'm at a loss to explain this dependency. Would you mind indulging my curiosity?

**Comment by Philippus** [comment, marked as answer]
That's for running the tests for scala.js.

You can use `parserCombinatorsJVM / test` in `sbt` to run just the JVM testsuite.

**Reply by peteraldous** [reply]
Thanks for the quick response!

**Reply by peteraldous** [reply]
It's worth adding, in case anyone else comes across this, that I had a lot of trouble using the library as a submodule because sbt insisted on running all tests when I tried to test the project that depended on this project. I ended up just symlinking/copying a jar built by doing `sbt +package` into the lib directory. It was a pain but it worked.
